### PR TITLE
feat(annotations): Don't show highlight button if no download permission

### PR DESCRIPTION
**Pull request by ConradJChan**

### CODE REVIEW

File: src/lib/viewers/doc/DocBaseViewer.js [hunk 1]
```diff
@@ -1105,11 +1105,13 @@ class DocBaseViewer extends BaseViewer {
         this.controls.add(__('exit_fullscreen'), this.toggleFullscreen, 'bp-exit-fullscreen-icon', ICON_FULLSCREEN_OUT);
 
         if (this.areNewAnnotationsEnabled() && this.hasAnnotationCreatePermission()) {
+            const canDownload = checkPermission(this.options.file, PERMISSION_DOWNLOAD);
```
Comment (jstoffan) [inline src/lib/viewers/doc/DocBaseViewer.js:1108]: Should we use the `canDownload` util from lib/file.js?

File: src/lib/viewers/doc/__tests__/DocBaseViewer-test.js [hunk 2]
```diff
@@ -2380,6 +2381,18 @@ describe('src/lib/viewers/doc/DocBaseViewer', () => {
                 }),
             );
 
+            it('should not showHighlightText if file has no download permission', () => {
+                stubs.checkPermission.withArgs(docBase.options.file, PERMISSION_DOWNLOAD).returns(false);
```
Comment (jstoffan) [inline src/lib/viewers/doc/__tests__/DocBaseViewer-test.js:2385]: Any way we can avoid using `withArgs` here? Will make converting to Jest (someday) easier.

File: src/lib/viewers/doc/DocBaseViewer.js (hunk 1 above)
Comment (ConradJChan) [reply src/lib/viewers/doc/DocBaseViewer.js:1108]: Seems like `canDownload` from `lib/file.js` checks a lot more than just the download permission related to actually being able to download the file. Might be too restrictive?

File: src/lib/viewers/doc/__tests__/DocBaseViewer-test.js (hunk 2 above)
Comment (ConradJChan) [reply src/lib/viewers/doc/__tests__/DocBaseViewer-test.js:2385]: I looked into this but it turns out `stubs.checkPermission` is already being used in the other tests

File: src/lib/viewers/doc/DocBaseViewer.js (hunk 1 above)
Comment (jstoffan) [reply src/lib/viewers/doc/DocBaseViewer.js:1108]: Yeah, fair point. It also looks like we only check the permission for `textLayerMode`. Should be good to go here.

**Review by jstoffan — APPROVED** [review]

**Pull request merged**